### DISCUSSION TITLE
is there a way to refresh data from postgresql to kong worker by hand?

### DISCUSSION THREAD

**Discussion by zffocussss** [Kong/kong#11968] [Help]
Hi team:
  I am using kong gateway + kong ingress controller + postgresql in AKS.
  does kong data(targets upstreams and so on) not keep sync with postgresql through a timer?If so,does kong provide a way to help user refresh it from postgresql by hand?
  as far as I know,kong will refresh its own cache from pqsql whenever k8s service endpoints are changed?
  what kong will happend if postgres server recover after it shutdown for some time?

**Comment by StarlightIbuki** [comment]
In this case, Kong uses Postgres as the ground truth of the config source and caches the queries to it. You can purge all the caches via the admin API (`DELETE /cache`).
But this is not recommended. Kong expects changes to the config are always through admin API (traditional) or config push (DBless, hybrid), and it relies on events triggered during the process to update some in-memory states. Direct manipulation of the database may corrupt the internal states of Kong and cause unexpected behavior.
When the database is down, Kong is not expected to work (for traditional mode). DP in a hybrid setup does not rely on a database thus not affected.
